apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/fakeResolver.kt

```diff
@@ -265,9 +265,39 @@ fun <T> buildData(
     resolver: FakeResolver,
 ): T {
   return adapter.obj(false).fromJson(
-      MapJsonReader(
-          buildFakeObject(selections, typename, map, resolver)
-      ),
-      CustomScalarAdapters.Unsafe
+      MapJsonReader(buildFakeObject(selections, typename, map, resolver)),
+      CustomScalarAdapters.PassThrough
+  )
+}
+
+fun <T> buildFragmentData(
+    adapter: Adapter<T>,
+    selections: List<CompiledSelection>,
+    typename: String,
+    block: Any? = null,
+    resolver: FakeResolver,
+    type: CompiledType
+): T {
+  val map = if (block == null) {
```

Comment:
This has to be a relatively different code path because fragments can be on interfaces types so we can't know in advance the builder to instanciate:

```kotlin
FragmentImpl.Data {
  // buildDroid is also an option
  buildHuman {
    ...
  }
}
```

```kotlin
QueryImpl.Data {
  // here the receiver is always QueryBuilder
  name = ...
}
```